lib/utils/temporal.mjs

```diff
@@ -0,0 +1,93 @@
+/**
+## mapp.utils.temporal
+
+The temporal utils module exports the temporal object with method to transform between unix date[time] integer and string values.
+
+@module /utils/temporal
+*/
+
+export const temporal = {
+  dateString,
+  timeString,
+  dateToUnixEpoch,
+  datetime,
+  date
+};
+
+/**
+@function dateString
+@description
+Entries of type date are stored as unix epoch. 
+This function extracts Date string from date entry integer value and returns it
+formatted according to local options passed inside entry.
+
+@param {infoj-entry} entry type:date or type:datetime infoj-entry typedef object.
+@property {integer} entry.value date[time] as unix epoch.
+@property {integer} [entry.newValue] updated entry.value which has not yet been saved.
+@returns {String} date with local format
+**/
+function dateString(entry) {
+  return new Date((entry.newValue || entry.value) * 1000)
+    .toLocaleDateString(entry.locale, entry.options)
+}
+
+/**
+@function timeString
+@description
+Extracts time string from unix epoch. Value is included in the final output of datetime entry formatter. 
+Returns time only as 00:00.
+
+@param {infoj-entry} entry type:date or type:datetime infoj-entry typedef object.
+@property {integer} entry.value date[time] as unix epoch.
+@property {integer} [entry.newValue] updated entry.value which has not yet been saved.
+@returns {String}
+*/
+function timeString(entry) {
+  return new Date((entry.newValue || entry.value) * 1000)
+    .toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' })
```

Comment:
<img width="916" alt="image" src="https://github.com/user-attachments/assets/938c6466-1730-48c2-9f43-ae42186ae1d8">

we are going to want to use the [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) object rather as this is going to be way more efficient.

Reply:
does it mean formatting by toLocaleString should be replaced entirely?

Reply:
Yeah! We did the same for the numeric formatter